pyartcd/pyartcd/pipelines/build_rhcos.py

```diff
@@ -58,184 +52,72 @@ def __init__(self, runtime: Runtime, new_build: bool, ignore_running: bool, vers
-            )
-            builds_info = response.json()["builds"]
-            for b in builds_info:
-                # build is still running when it has no status
-                if b["result"] is None:
-                    build = dict(**b, job=job, parameters=self.build_parameters(b), url=self.build_url(job, b["number"]))
-                    builds.append(build)
-
-        return [b for b in builds if b["parameters"].get("STREAM") == self.stream]
-
-    @property
-    def stream(self):
-        if self._stream:
-            return self._stream
-
-        # doozer --quiet -g openshift-4.14 config:read-group urls.rhcos_release_base.multi --default ''
-        # https://releases-rhcos-art.apps.ocp-virt.prod.psi.redhat.com/storage/prod/streams/4.14-9.2/builds
-        cmd = [
-            "doozer",
-            "--quiet",
-            "--group", f'openshift-{self.version}',
-            "config:read-group",
-            "urls.rhcos_release_base.multi",
-            "--default",
-            "''"
-        ]
-        result = subprocess.run(cmd, stdout=PIPE, stderr=PIPE, check=False, universal_newlines=True)
-        if result.returncode != 0:
-            raise IOError(f"Command {cmd} returned {result.returncode}: stdout={result.stdout}, stderr={result.stderr}")
-        match = re.search(r'streams/(.*)/builds', result.stdout)
-        if match:
-            self._stream = match[1]
-        else:
-            self._stream = self.version
-        return self._stream
+        """Check if there are any existing builds for the given job with given version. Returns builds dicts."""
+        if self.ignore_running:
+            return []
+        response = self.request_session.get(
+            f"{JENKINS_BASE_URL}/job/{self.job}/api/json?tree=builds[number,description,url,result,actions[parameters[name,value]]]",
+        )
+        builds_info = response.json()["builds"]
+        return [{"result": None, "description": build["description"], "url": build["url"]}
+                for build in builds_info
+                if build["result"] is None
+                and any(
+                    param["name"] == "STREAM" and param["value"] == self._stream  # check build parameter with given stream
+                    for action in build["actions"]
+                    if "parameters" in action
+                    for param in action["parameters"]
+        )]
+
+    def get_stream(self):
+        """Get rhcos job's stream parameter value, for 4.12+ it looks like 4.x-9.x for 4.12 is just 4.12"""
+        group_response = self.request_session.get(f"{OCP_REPO_RAW}/openshift-{self.version}/group.yml")
+        group_file = yaml.load(group_response.text, Loader=yaml.SafeLoader)
+        return f"{self.version}-{group_file['vars']['RHCOS_EL_MAJOR']}.{group_file['vars']['RHCOS_EL_MINOR']}" if group_file['vars']['MINOR'] > 12 else self.version
```

Comment:
Could you make an assertion about ``MAJOR` as well, so this does not break when we start openshift5?

Reply:
do you mean MAJOR <= 4 ?